docs/DeveloperGuide.md

```diff
@@ -1,13 +1,249 @@
 
+### Storage Component
+
+API: `Storage.java`
+
+![](images/Storage.png)
+
+* The storage component loads data from the saved text files when the application starts, and saves the data to the
+  text files when the application exits.
+* The storage class uses the static methods in LoadData and SaveData to load and save data respectively.
+* The `load` method in LoadData reads the `data.txt` file and loads any existing Income, Expense and Budget into the application.
+* The `save` method in SaveData saves all Incomes, Expenses and existing Budget into the `data.txt` file.
+
+#### Design considerations:
+
+* There are 2 main ways to implement the storage, one is to save the data after every command, and the other is to save 
+the data one upon exiting the program with the `exit` command.
+* Saving the data once upon exit (Currently implemented):
+  * Advantage: Better efficiency and performance of the program.
+  * Disadvantage: If the program crashes or exits incorrectly, data will not be saved.
+* Saving the data after every command:
+  * Advantage: Changes are saved after every command.
+  * Disadvantage: Executing command might slow down the program when there is a large amount of data to be saved.
+
+### Visualization Feature 
+
+This feature is implemented with the help of [XChart](https://knowm.org/open-source/xchart/), a simple charting library for Java by Knowm.
+
+By typing in the vis command with the appropriate arguments (/s and /t), users will be able to visualize their income or expense 
+using visualization tools (Piechart, Bar Chart)
+
+Demo: 
+
+`vis /t expense /c pie`
+
+Output
+
+`Displaying piechart for expense`
+A message will be shown telling you that the chart is being displayed
+
+![](images/vis/visOutput.png)
+
+This feature was implemented with the help of three different classes.
+They are namely: Visualizer, Categorizer, VisCommand (Inherits from abstract Command Class)
+
+VisCommand's Role: 
+1) Read the parameters of the vis command entered by the user
+- `/t` Reads the type of cashflow that the user wants to visualize (income/expense)
+- `/c` Reads the type of visualization tools the user wants (piechart/barchart)
+
+2) Calls the Cateorgizer to sort cashflow (Income/Expense) according to type
+
+3) Calls the Visualizer to display the chart to the user
+
+Categorizer's Role: 
+
+According to the cashflow type (Income/Expense) arugment passed in, the Categorizer sorts the 
+specified cashflow entry according to type using a Hashmap which is returned and used by the Visualizer
+
+Visualizer's Role: 
+
+According to the chart type (Pie/Bar) argument and the Hashmap obtained from the categorizer passed in, 
+the visualizer displays the specified visualization chart by calling the charting library Xchart.
+
+### Class Diagram
+
+![](images/vis/visualisationClass.png)
+
+### Sequence Diagram 
+
+Overall 
+
+![](images/vis/visualisationSequence.png)
```

Comment:
Maybe you could include return arrows. Also it could perhaps be helpful to make the alt and opt boxes coherent by applying the coloring to all sequence diagrams in the DG (helps create impression that DG was written by a single person).